components/profile.js

```diff
@@ -65,57 +67,413 @@ SteamCommunity.prototype.editProfile = function(settings, callback) {
-					values.country = settings[i];
+					out[10] = {
+						"name": formd[10].name,
+						"value": settings[i]
+					};
 					break;
 
 				case 'state':
-					values.state = settings[i];
+					out[11] = {
+						"name": formd[11].name,
+						"value": settings[i]
+					};
 					break;
 
 				case 'city':
-					values.city = settings[i];
+					out[12] = {
+						"name": formd[12].name,
+						"value": settings[i]
+					};
 					break;
 
 				case 'customURL':
-					values.customURL = settings[i];
+					out[13] = {
+						"name": formd[13].name,
+						"value": settings[i]
+					};
 					break;
 
 				case 'background':
 					// The assetid of our desired profile background
-					values.profile_background = settings[i];
+					out[15] = {
+						"name": formd[15].name,
+						"value": settings[i]
+					};
 					break;
 
 				case 'featuredBadge':
 					// Currently, game badges aren't supported
-					values.favorite_badge_badgeid = settings[i];
+						out[16] = {
+							"name": formd[16].name,
+							"value": settings[i]
+						};
 					break;
 
 				case 'primaryGroup':
-					if(typeof settings[i] === 'object' && settings[i].getSteamID64) {
-						values.primary_group_steamid = settings[i].getSteamID64();
+					if (typeof settings[i] === 'object' && settings[i].getSteamID64) {
+						out[18] = {
+							"name": formd[18].name,
+							"value": settings[i].getSteamID64()
+						};
 					} else {
-						values.primary_group_steamid = new SteamID(settings[i]).getSteamID64();
+						out[18] = {
+							"name": formd[18].name,
+							"value": new SteamID(settings[i]).getSteamID64()
+						};
 					}
 
 					break;
 
-				// TODO: profile showcases
+				case 'showcases':
+					for (var t in settings[i]) {
+						var showcaseconfig = [false,0,0,[]];
+
+						switch (settings[i][t].showcase) {
+
+							case 'infobox':
+								out.push({
+									"name": "profile_showcase[]",
+									"value": "8"
+								});
+								out.push({
+									"name": "rgShowcaseConfig[8][0][title]",
+									"value": settings[i][t]["values"]["title"]
+								});
+								out.push({
+									"name": "rgShowcaseConfig[8][0][notes]",
+									"value": settings[i][t]["values"]["notes"]
+								});
+								break;
+
+							case 'artwork':
+								out.push({
+									"name": "profile_showcase[]",
+									"value": "13"
+								});
+								for (var n = 0; n < 4; n++) {
+									out.push({
+										"name": "rgShowcaseConfig[13][" + n + "][publishedfileid]",
+										"value": settings[i][t]["values"][n] || ""
+									});
+								}
+								break;
+
+							case 'trade':
+								out.push({
+									"name": "profile_showcase[]",
+									"value": "4"
+								});
+								out.push({
+									"name": "rgShowcaseConfig[4][6][notes]",
+									"value": settings[i][t]["values"]["notes"]
+								});
+
+								if (settings[i][t]["values"].hasOwnProperty("items")) {
+									showcaseconfig = [true, 6, 4, settings[i][t]["values"]["items"]];
```

Comment:
How does this work?

Reply:
Only if items are submitted the requests to change single games / items / ... are done, this is a simple check to do that.